### PR TITLE
feat(Bonus Pagamenti Digitali): [#176072451] Added "share you screenshot" checkbox to "Write to the IO team" screen

### PR DESCRIPTION
**Pull request by DavideValdo**
## Short description
A checkbox to express one's will to also attach a screenshot has been added to the "Write to the IO team" screen. 

## How to test
- Click the ? sign on almost any screen
- Scroll through the text and click "Write to the IO team"
- Scroll through the end and see the checkbox

Schermata 2020-12-09 alle 19 51 02

### CODE REVIEW

File: ts/components/SendSupportRequestOptions.tsx [hunk 1]
```diff
@@ -106,16 +118,37 @@ const SendSupportTokenInfo: React.FunctionComponent<Props> = ({
...
       </Content>
       <FooterWithButtons
         type={"SingleButton"}
-        leftButton={continueButtonProps(() => {
-          onContinue(sendPersonalInfo);
-        })}
+        leftButton={continueButtonProps(() =>
+          onContinue({
+            sendPersonalInfo,
+            sendScreenshot: sendScreenshot as boolean
```
Comment (Undermaken) [inline ts/components/SendSupportRequestOptions.tsx:146]:
```suggestion
            sendScreenshot
```

File: ts/components/ContextualHelpModal.tsx [hunk 2]
```diff
@@ -182,10 +184,11 @@ const ContextualHelpModal: React.FunctionComponent<Props> = (props: Props) => {
     >
       <Container>
         {showSendPersonalInfo ? (
-          <SendSupportTokenInfo
+          <SendSupportRequestOptions
+            initialScreenshotCheckboxValue={false}
```
Comment (Undermaken) [inline ts/components/ContextualHelpModal.tsx:188]:
```suggestion
```
For now it's better to keep the default behavior: don't ask about screenshot sharing.

File: ts/components/SendSupportRequestOptions.tsx [hunk 3]
```diff
@@ -19,7 +19,8 @@ import { IOStyles } from "./core/variables/IOStyles";
 type Props = {
   onClose: () => void;
   onGoBack: () => void;
-  onContinue: (sharingPersonalData: boolean) => void;
+  onContinue: (options: SupportRequestOptions) => void;
+  initialScreenshotCheckboxValue: boolean | undefined;
```
Comment (Undermaken) [inline ts/components/SendSupportRequestOptions.tsx:23]:
```suggestion
  initialScreenshotCheckboxValue?: boolean;
```
in this way you can avoid to define this property when you're using this component

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/components/SendSupportRequestOptions.tsx [hunk 4]
```diff
@@ -106,16 +118,37 @@ const SendSupportTokenInfo: React.FunctionComponent<Props> = ({
...
           </View>
         </View>
+        {initialScreenshotCheckboxValue !== undefined && (
+          <View style={styles.checkBoxContainer}>
+            <RawCheckBox
+              checked={sendScreenshot}
+              onPress={() => setSendScreenshot(ov => !ov)}
+            />
+            <View hspacer={true} />
+            <View style={{ flex: 1 }}>
```
Comment (debiff) [inline ts/components/SendSupportRequestOptions.tsx:128]: it is better to use:
`<View style={IOStyles.flex}>`

**Review by debiff — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]

**Review by debiff — APPROVED** [review]

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**